web/modules/custom/dpl_graphql/dpl_graphql.install

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @file
+ * Install, update and uninstall functions for the dpl_graphql module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\drupal_typed\DrupalTyped;
+use Drupal\simple_oauth\Service\Exception\ExtensionNotLoadedException;
+use Drupal\simple_oauth\Service\Exception\FilesystemValidationException;
+use Drupal\simple_oauth\Service\KeyGeneratorService;
+
+/**
+ * Run on initial site setup.
+ *
+ * We want to run the dpl_graphql_update_10001 on both
+ * new and existing sites.
+ */
+function dpl_graphql_install(): string {
+  $messages[] = dpl_graphql_update_10001();
+
+  return implode('\r\n', $messages);
+
+}
+
+/**
+ * Generates public and private keys used for the simple_oauth module.
+ */
+function dpl_graphql_update_10001(): string {
+  $file_system = DrupalTyped::service(FileSystemInterface::class, 'file_system');
+  $key_generator = DrupalTyped::service(KeyGeneratorService::class, 'simple_oauth.key.generator');
+
+  $path = '../web/sites/default/files/simple_oauth_keys';
+  $file_system->prepareDirectory($path, FileSystemInterface::CREATE_DIRECTORY);
```

Comment:
The files need to be written to a private directory so the are not accessible publicly.
The best think would be that we could write in a mount point that is not accesible at all by the web.
We could look into that later.
But for now we need to make sure that the directory that we create in `sites/default/files` is protected.
You need to investigate if the path to the private dir is set in settings, otherwise you shoul be able to set it via:
```
$settings['file_private_path'] = SOME_ABSOLUTE_PATH_TO_THE_DIRECTORY
````
When you need the path to the dir you should be able to do so by calling:
```
$private_files_path = \Drupal::service('file_system')->realpath('private://');